internal/egress/egress.go

```diff
@@ -113,29 +98,36 @@ func DialSSH(channel ssh.Channel, bc BackendConn) error {
 	return nil
 }
 
-// ParseBackendInfo takes a array containing our payload command and returns
+// ParseBackendInfo takes a string containing our payload command and returns
 // a BackendConn struct with the required infos to call DialSSH
 func ParseBackendInfo(payload string) (bc BackendConn, err error) {
+	if len(payload) == 0 || len(payload) > 1024 {
+		return bc, errors.New("Invalid payload")
+	}
+
+	//Remove leading and trailing whitespaces
+	payload = strings.TrimSpace(payload)
+
 	command := strings.Split(payload, " ")
 
-	if command == nil || len(command) < 1 {
+	if command == nil || len(command) < 2 {
 		return bc, errors.New("Invalid payload")
 	}
 
-	reg := regexp.MustCompile(`^.+@.+$`)
-	regPort := regexp.MustCompile(`-p`)
-
-	for i, s := range command {
-		if reg.Match([]byte(s)) {
-			arr := strings.Split(s, `@`)
+	c := command[0]
 
-			if net.ParseIP(arr[1]) == nil {
-				return bc, errors.New("Could not parse host")
-			}
+	if c == "ssh" {
+		bc.Command = "ssh"
+	} else if c == "telnet" {
+		bc.Command = "telnet"
+	} else if c == "bastion" {
+		bc.Command = "bastion"
+	}
```

Comment:
It should return an error if the command is different from ssh, telnet and bestion